textproto/header.go

```diff
@@ -421,18 +421,37 @@ func trimAroundNewlines(v []byte) string {
 }
 
 const (
-	maxHeaderLines = 1000
-	maxLineOctets  = 4000
+	defaultMaxLineOctets = 4000
+	maxHeaderLines       = 1000
 )
 
+// ReadOpts represents the options in a header reader.
+//
+// The supplied options to the ReadHeader can be nil. In this case the default
+// value for MaxLineOctets will be set to the value of defaultMaxLineOctets constant.
+type ReadOpts struct {
+	MaxLineOctets int
```

Comment:
Maybe we should improve the name of this field.

- "Octet" is often used in RFCs, less so in Go APIs. We should probably use "bytes" or "len"
- We should make it clear this only applies to header lines
- Should we expose a limit for the total header field size, instead of a per-line limit?

Reply:
>"Octet" is often used in RFCs, less so in Go APIs. We should probably use "bytes" or "len"
>We should make it clear this only applies to header lines

MaxHeaderLineLength? Seems wordy.
MaxHdrLineBytes? 

>Should we expose a limit for the total header field size, instead of a per-line limit? In the end that's what matters, or am I missing something?

In my PR, I also added maxHeaderLines. The goal was to limit overall header size. Having it capped at maxHeaderLines*maxLineOctets (currently ~3.8 MiB)

Reply:
`MaxHeaderLineLength` SGTM.